discovery/syncer_test.go

```diff
@@ -769,14 +780,45 @@ func TestGossipSyncerReplyChanRangeQuery(t *testing.T) {
 
 			// If this is not the last chunk, then Complete should
 			// be set to zero. Otherwise, it should be one.
+			offset := 0
```

Comment:
nit: comment should be moved down? also add rationale for why offset is set this way?